core/vm/gas_table.go

```diff
@@ -55,6 +55,37 @@ func memoryGasCost(mem *Memory, newMemSize uint64) (uint64, error) {
 	return 0, nil
 }
 
+// memoryGasCost calculates the quadratic gas for memory expansion. It does so
+// only for the memory region that is expanded, not the total memory.
+// doesn't modify the provided mem object
+func MemoryGasCost(mem *Memory, newMemSize uint64) (uint64, error) {
```

Comment:
Simpler: 
````
var MemoryGasCost = memoryGasCost
```
But I don't actually think we should/need to expose this

Reply:
the reason I have a separate function here is that I didn't want the tracer to mutate the `Memory` object when it calculates the memory expansion gas fee.  kinda moot though given the point I wrote above.